pontos/github/argparser.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (C) 2022 Greenbone Networks GmbH
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+""" Argument parser for pontos-github """
+
+from argparse import ArgumentParser, Namespace
+import os
+from typing import List
+
+
+def from_env(name: str) -> str:
+    if name in os.environ:
+        return os.environ[name]
+    else:
+        return name
+
+
+def parse_args(
+    args: List[str] = None,
+) -> Namespace:
+    """Parsing args for nasl-lint
+
+    Arguments:
+    args        The programm arguments passed by exec
+    term        The terminal to print"""
+
+    parser = ArgumentParser(
+        description="Greenbone GitHub API.",
+    )
+
+    subparsers = parser.add_subparsers(
+        title='subcommands',
+        description='valid subcommands',
+        help='additional help',
+        dest='command',
+    )
+
+    # create a PR from command line
+    pr_parser = subparsers.add_parser(
+        'pull-request', aliases=['pr', 'PR', 'pullrequest']
+    )
+
+    pr_parser.add_argument(
+        "repo", help=("GitHub repository (owner/name) to use")
+    )
+
+    pr_parser.add_argument(
+        "head",
+        help=("Branch to create a pull request from"),
+    )
+
+    pr_parser.add_argument(
+        "target",
+        default="main",
+        help=("Branch as as target for the pull"),
+    )
+
+    pr_parser.add_argument(
+        "title",
+        help=("Title for the pull request"),
+    )
+
+    pr_parser.add_argument(
+        "-b",
+        "--body",
+        default="#empty body",
```

Comment:
Personally I would require the description. The user could still pass `--body ""`
```suggestion
        required=True,
```

Reply:
How about using the `PULL_REQUEST_TEMPLATE.md` as default?
Another Idea that came in my mind: Using distinct parameters for `--why` `--what` and `--how`?